incubator/kafka/templates/kafka-petset.yaml

```diff
@@ -0,0 +1,94 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: "{{ printf "%s-%s-broker" .Release.Name .Values.Name | trunc 24 }}"
+  labels:
+    heritage: {{.Release.Service | quote }}
+    release: {{.Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+  annotations:
+    "helm.sh/created": {{.Release.Time.Seconds | quote }}
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+spec:
+  ports:
+  - port: 9092
+  clusterIP: None
+  selector:
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+    app: kafka
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+  labels:
+    heritage: {{.Release.Service | quote }}
+    release: {{.Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+  annotations:
+    "helm.sh/created": {{.Release.Time.Seconds | quote }}
+spec:
+  ports:
+  - port: 9092
+  selector:
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+    app: kafka
+---
+apiVersion: apps/v1alpha1
+kind: PetSet
+metadata:
+  name: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+  labels:
+    app: kafka
+    heritage: {{.Release.Service | quote }}
+    release: {{.Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+  annotations:
+    "helm.sh/created": {{.Release.Time.Seconds | quote }}
+spec:
+  serviceName: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+  replicas: {{ default 3 .Values.Replicas }}
+  template:
+    metadata:
+      labels:
+        component: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+        pod.alpha.kubernetes.io/init-containers: '[
+        ]'
+    spec:
+      containers:
+      - name: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+        image: "{{.Values.Image}}:{{.Values.ImageTag}}"
+        imagePullPolicy: "{{.Values.ImagePullPolicy}}"
+        ports:
+        - containerPort: 9092
+          name: kafka
+        resources:
+          requests:
+            cpu: "{{ .Values.Cpu }}"
+            memory: "{{ .Values.Memory }}"
+        command:
+        - sh
+        - -c
+        - "./bin/kafka-server-start.sh config/server.properties --override zookeeper.connect={{ printf "%s-%s" .Release.Name .Values.zookeeper.Name | trunc 24 }}:2181/ --override broker.id=${HOSTNAME##*-}"
+        volumeMounts:
+        - name: datadir
+          mountPath: /tmp/zookeeper
+      volumes:
+      - name: workdir
+        emptyDir: {}
+  volumeClaimTemplates:
+  - metadata:
+      name: datadir
+      annotations:
+        volume.alpha.kubernetes.io/storage-class: anything
```

Comment:
Should we have this setup to do both a PVC or an empty dir?  Looking for an example ...